mongodump_passthrough/functions.yml

```diff
@@ -0,0 +1,773 @@
+    command: expansions.write
+    params:
+      file: expansions.yml
+      redacted: true
+
+  # Changed for mongodump_passthrough
+  f_repo_fetch: &f_repo_fetch
+    command: git.get_project
+    params:
+      directory: src/github.com/mongodb/mongo-tools
+      revisions:
+        mongosync: *_mongosync_pin
+
+  # Added for mongodump_passthrough
+  f_link_to_mongo_tools: &f_link_to_mongo_tools
+    command: shell.exec
+    params:
+      shell: bash
+      script: |
+        ln -s ${workdir}/src/github.com/mongodb/mongo-tools src
+
+  "set up logger credentials": &f_set_up_logger_credentials
+    command: subprocess.exec
+    params:
+      binary: "./src/mongosync/evergreen/logger_credentials_setup.sh"
+      add_expansions_to_env: true
+
+  f_resmoke_wheelhouse_setup: &f_resmoke_wheelhouse_setup
+    command: subprocess.exec
+    params:
+      binary: "./src/mongosync/evergreen/resmoke_wheelhouse_setup.sh"
+      add_expansions_to_env: true
+
+  # Function used to download particular bits of resmoke and binaries uploaded previously.
+  f_resmoke_with_binaries_fetch: &f_resmoke_with_binaries_fetch
+    command: s3.get
+    params:
+      aws_key: ${aws_key}
+      aws_secret: ${aws_secret}
+      # Changed for mongodump_passthrough
+      remote_file: mongo-tools/mongodump_passthrough/${build_variant}/${revision}/resmoke-with-binaries/${build_id}.tgz
+      bucket: mciuploads
+      extract_to: *_resmoke_dir
+
+  f_resmoke_wheelhouse_fetch: &f_resmoke_wheelhouse_fetch
+    command: s3.get
+    params:
+      aws_key: ${aws_key}
+      aws_secret: ${aws_secret}
+      # Changed for mongodump_passthrough
+      remote_file: mongo-tools/mongodump_passthrough/${build_variant}/${revision}/resmoke-python-wheelhouse/${build_id}.tgz
+      bucket: mciuploads
+      extract_to: "."
+
+  f_migration_verifier_binary_fetch: &f_migration_verifier_binary_fetch
+    command: s3.get
+    params:
+      aws_key: ${aws_key}
+      aws_secret: ${aws_secret}
+      # Changed for mongodump_passthrough
+      remote_file: mongo-tools/mongodump_passthrough/${mongosync_compile_build_variant}/${revision}/${version_id}/migration_verifier
+      bucket: mciuploads
+      local_file: src/mongosync/migration_verifier
+
+  f_make_migration_verifier_binary_executable:
+    &f_make_migration_verifier_binary_executable
+    command: shell.exec
+    params:
+      shell: bash
+      working_dir: src/mongosync
+      script: |
+        set -o errexit
+        set -o pipefail
+        set -o xtrace
+
+        chmod +x migration_verifier
+
+  f_mongosync_binary_fetch: &f_mongosync_binary_fetch
+    command: s3.get
+    params:
+      aws_key: ${aws_key}
+      aws_secret: ${aws_secret}
+      # Changed for mongodump_passthrough
+      remote_file: mongo-tools/mongodump_passthrough/${mongosync_compile_build_variant}/${revision}/${version_id}/${mongosync_binary_folder}/${version_id}.tgz
+      bucket: mciuploads
+      extract_to: "src/mongosync"
```

Comment:
Do you need to do this at all? Do you actually run `mongosync` for tools passthroughs?

Reply:
I don't build or run any mongosync binaries.  But I copy the mongodump and mongorestore binaries into the mongosync/dist directory, they get packed up and sent to s3 from there, and then they get downloaded back into mongosync/dist.

So we have the same task names and same dependencies as for mongosync passthrough tests, we're just gluing mongodump and mongorestore into that same framework.

Reply:
I think a comment explaining this would be good.

Reply:
Added a comment